# Genomic Disease Prediction using Polygenic Risk Scores (PRS)

The project mainly focuses on evaluating machine learning models and predicting specific genetically inherited diseases using **Polygenic Risk Scores (PRS)**, leveraging Genomic data and machine learning techniques.
The work was carried out as a part of an internship at **IIT Guwahati** .

## ðŸ”¬ Project Overview

The goal of this project is to assess the predicitive capability of **common SNPs (Single Nucleotide Polymorphism)** using PRS, applied across multiple diseases:
- **Rheumatoid Arthritis**
- **Alzheimer's Disease**
- **Type 2 Diabetes**
- **Hyperthyroidism**

## ðŸ“ Project Structure

```
Internship_Project/
â”œâ”€ Final Datasets/
â”‚  â”œâ”€ ALZ_final/
â”‚  â”‚  â”œâ”€ ALZ_aligned_genotypes_final.csv
â”‚  â”‚  â”œâ”€ ALZ_aligned_phenotypes_final_extended.csv
â”‚  â”‚  â””â”€ ALZ_aligned_phenotypes_final.csv
â”‚  â”œâ”€ HYP_final/
â”‚  â”‚  â”œâ”€ HYP_aligned_genotypes_final.csv
â”‚  â”‚  â”œâ”€ HYP_aligned_phenotypes_final_extended.csv
â”‚  â”‚  â””â”€ HYP_aligned_phenotypes_final.csv
â”‚  â”œâ”€ RA_final/
â”‚  â”‚  â”œâ”€ RA_aligned_genotypes_final.csv
â”‚  â”‚  â”œâ”€ RA_aligned_phenotypes_final_extended.csv
â”‚  â”‚  â””â”€ RA_aligned_phenotypes_final.csv
â”‚  â””â”€ T2D_final/
â”‚     â”œâ”€ T2D_aligned_genotypes_final.csv
â”‚     â”œâ”€ T2D_aligned_phenotypes_final_extended.csv
â”‚     â””â”€ T2D_aligned_phenotypes_final.csv
â”œâ”€ Models/
â”‚  â”œâ”€ Alzheimers.ipynb
â”‚  â”œâ”€ Hyperthyroidism.ipynb
â”‚  â”œâ”€ Rheumatoid_Arhtiritis.ipynb
â”‚  â””â”€ Type_2_diabetes.ipynb
â”œâ”€ Outputs/
â”‚  â”œâ”€ ALZ_results/
â”‚  â”‚  â”œâ”€ ALZ_confusion_matrix.png
â”‚  â”‚  â”œâ”€ ALZ_PRS_distribution_fixed.png
â”‚  â”‚  â”œâ”€ ALZ_prs_scores.csv
â”‚  â”‚  â”œâ”€ ALZ_selectKbest_logreg_model_AgeSex.joblib
â”‚  â”‚  â”œâ”€ ALZ_selectKbest_logreg_model.joblib
â”‚  â”‚  â”œâ”€ ALZ_selectKbest_model_performance_AgeSex.png
â”‚  â”‚  â”œâ”€ ALZ_selectKbest_model_performance.png
â”‚  â”‚  â”œâ”€ logreg_selectKbest_coefficients_AgeSex.csv
â”‚  â”‚  â””â”€ logreg_selectKbest_coefficients.csv
â”‚  â”œâ”€ HYP_results/
â”‚  â”‚  â”œâ”€ HYP_confusion_matrix.png
â”‚  â”‚  â”œâ”€ HYP_PRS_distribution_fixed.png
â”‚  â”‚  â”œâ”€ HYP_prs_scores.csv
â”‚  â”‚  â”œâ”€ HYP_selectKbest_logreg_model_AgeSex.joblib
â”‚  â”‚  â”œâ”€ HYP_selectKbest_logreg_model.joblib
â”‚  â”‚  â”œâ”€ HYP_selectKbest_model_performance_AgeSex.png
â”‚  â”‚  â”œâ”€ HYP_selectKbest_model_performance.png
â”‚  â”‚  â”œâ”€ logreg_selectKbest_coefficients_AgeSex.csv
â”‚  â”‚  â””â”€ logreg_selectKbest_coefficients.csv
â”‚  â”œâ”€ RA_results/
â”‚  â”‚  â”œâ”€ logreg_selectKbest_coefficients_AgeSex.csv
â”‚  â”‚  â”œâ”€ logreg_selectKbest_coefficients.csv
â”‚  â”‚  â”œâ”€ RA_confusion_matrix.png
â”‚  â”‚  â”œâ”€ RA_PRS_distribution_fixed.png
â”‚  â”‚  â”œâ”€ RA_prs_scores.csv
â”‚  â”‚  â”œâ”€ RA_selectKbest_logreg_model_AgeSex.joblib
â”‚  â”‚  â”œâ”€ RA_selectKbest_logreg_model.joblib
â”‚  â”‚  â”œâ”€ RA_selectKbest_model_performance_ageSex.png
â”‚  â”‚  â””â”€ RA_selectKbest_model_performance.png
â”‚  â”œâ”€ T2D_results/
â”‚  â”‚  â”œâ”€ logreg_selectKbest_coefficients_AgeSex.csv
â”‚  â”‚  â”œâ”€ logreg_selectKbest_coefficients.csv
â”‚  â”‚  â”œâ”€ T2D_confusion_matrix.png
â”‚  â”‚  â”œâ”€ T2D_PRS_distribution_fixed.png
â”‚  â”‚  â”œâ”€ T2D_prs_scores.csv
â”‚  â”‚  â”œâ”€ T2D_selectKbest_logreg_model_AgeSex.joblib
â”‚  â”‚  â”œâ”€ T2D_selectKbest_logreg_model.joblib
â”‚  â”‚  â”œâ”€ T2D_selectKbest_model_performance_AgeSex.png
â”‚  â”‚  â””â”€ T2D_selectKbest_model_performance.png
â”‚  â””â”€ Table.ipynb
â”œâ”€ Raw_Datasets/
â”‚  â”œâ”€ ALZ_data/
â”‚  â”‚  â”œâ”€ ALZ_chr_dosage_matrix.csv
â”‚  â”‚  â”œâ”€ ALZ_chr_genotypes_raw.txt
â”‚  â”‚  â”œâ”€ ALZ_chr_genotypes_with_header.tsv
â”‚  â”‚  â”œâ”€ ALZ_top2000_cleaned.csv
â”‚  â”‚  â”œâ”€ phenotype_ALZ_aligned.csv
â”‚  â”‚  â”œâ”€ phenotype_ALZ_simulated_50_50.tsv
â”‚  â”‚  â””â”€ sample_ids(1).txt
â”‚  â”œâ”€ HYP_data/
â”‚  â”‚  â”œâ”€ HYP_chr_dosage_matrix.csv
â”‚  â”‚  â”œâ”€ HYP_chr_genotypes_raw.txt
â”‚  â”‚  â”œâ”€ HYP_chr_genotypes_with_header.tsv
â”‚  â”‚  â”œâ”€ HyperTH_top2000_cleaned.csv
â”‚  â”‚  â”œâ”€ phenotype_HYP_aligned.csv
â”‚  â”‚  â”œâ”€ phenotype_HYP_simulated_50_50.tsv
â”‚  â”‚  â””â”€ sample_ids(3).txt
â”‚  â”œâ”€ RA_data/
â”‚  â”‚  â”œâ”€ phenotype_RA_aligned.csv
â”‚  â”‚  â”œâ”€ phenotype_RA_simulated_50_50.tsv
â”‚  â”‚  â”œâ”€ RA_aligned_genotypes_final.csv
â”‚  â”‚  â”œâ”€ RA_aligned_phenotypes_final.csv
â”‚  â”‚  â”œâ”€ RA_chr1_6_dosage_matrix.csv
â”‚  â”‚  â”œâ”€ RA_chr1_6_genotypes_raw.txt
â”‚  â”‚  â”œâ”€ RA_chr1_6_genotypes_with_header.tsv
â”‚  â”‚  â”œâ”€ RA_top10000_cleaned.csv
â”‚  â”‚  â””â”€ sample_ids.txt
â”‚  â”œâ”€ T2D_data/
â”‚  â”‚  â”œâ”€ phenotype_T2D_aligned.csv
â”‚  â”‚  â”œâ”€ phenotype_T2D_simulated_50_50.tsv
â”‚  â”‚  â”œâ”€ sample_ids(2).txt
â”‚  â”‚  â”œâ”€ T2D_chr_dosage_matrix.csv
â”‚  â”‚  â”œâ”€ T2D_chr_genotypes_raw.txt
â”‚  â”‚  â”œâ”€ T2D_chr_genotypes_with_header.tsv
â”‚  â”‚  â””â”€ T2D_top2000_cleaned.csv
â”‚  â”œâ”€ Dataset_processing_ALZ.ipynb
â”‚  â”œâ”€ Dataset_processing_HYP.ipynb
â”‚  â”œâ”€ Dataset_processing_RA.ipynb
â”‚  â””â”€ Dataset_processing_T2D.ipynb
â”œâ”€ .gitignore
â”œâ”€ environment.yaml
â”œâ”€ LICENSE
â”œâ”€ output.md
â”œâ”€ output.txt
â””â”€ README.md
```

## ðŸ§¬ Datasets Used

- **1000 Genomes Project** (genotype data)
- Phenotype labels were **simulated** to follow disease-specific prevalance are used for binary classificaton.
- Two splits were tested:
  - **50/50 Case-Control**
  - **10/90 Case-Control** *(produced more realistic and better-performing results)*

## ðŸ§  Methodology

- **Dataset Collection**:
  - Used **GWAS Catalog** for SNP collection for the four diseases and **1000 Genomes Project** for chromosomes related to each particular trait
  - Used **pandas** to remove the NULL values and taking only the necessary columns *(rs_id, chrom, pos, p_value, beta)*

- **Genotype Preprocessing**:
  - Used **bcftools** to filter SNPs based on:
    - `MAF > 0.001`
    - `Missing Rate < 3%`
    - `Hardy-Weinberg Equillibrium (HWE) > 1e-6`
  - SNPs encoded as dosage values (0, 1, 2)

- **Feature Selection**:
  - GWAS p-value filtering (`p < 1e-5`)
  - `SelectKBest` using chi-square or ANOVA F-score

- **Modelling**:
  - **Logistic Regression (Lasso)**
  - Scoring Metrics: AUC, sensitivity, specificity
  - PRS calculated as a weighted sum:
    - PRSáµ¢ = Î£â±¼=1áµ (Î²â±¼ Ã— dosageáµ¢â±¼)

- **Evaluation**:
  - ROC AUC range for SNP-only models *(50/50 split)*:
    - RA: ~0.55
    - ALZ: ~0.60
    - T2D: ~ 0.53
    - HYP: ~0.53
  - ROC AUC range for SNP-only models *(10/90 split)*:
    - RA: ~0.60
    - ALZ: ~0.64
    - T2D: ~ 0.62
    - HYP: ~0.52

## âš’ï¸ Tools and Libraries

- Python (pandas, scikit-learn, matplotlib, numpy)
- bcftools
- Jupyter Notebook

## ðŸ”— Linked Resources

- [1000 Genomes Project](https://www.internationalgenome.org/)
- [GWAS Catalog](https://www.ebi.ac.uk/gwas/)
- [bcftools](https://github.com/samtools/bcftools)

## ðŸ™Œ Acknowledgments

- This project was completed under the guidance of **Prof. M.K. Bhuyan**, Department of EEE, **IIT Guwahati**.
- Thanks to [Pragyan Thapa](https://github.com/pragyanthapa) for collaborating on model development and experiments.
